app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/DeclineIncomingCallReceiver.kt

```diff
@@ -56,35 +57,32 @@ class DeclineIncomingCallReceiver : BroadcastReceiver() { // requires zero argum
     @ApplicationScope
     lateinit var coroutineScope: CoroutineScope
 
+    @Inject
+    lateinit var callNotificationManager: CallNotificationManager
+
     override fun onReceive(context: Context, intent: Intent) {
-        val conversationId: String = intent.getStringExtra(EXTRA_CONVERSATION_ID) ?: return
-        appLogger.i("CallNotificationDismissReceiver: onReceive, conversationId: $conversationId")
+        val conversationIdString: String = intent.getStringExtra(EXTRA_CONVERSATION_ID) ?: return
+        appLogger.i("CallNotificationDismissReceiver: onReceive, conversationId: ${conversationIdString.obfuscateId()}")
         coroutineScope.launch(Dispatchers.Default) {
-            val userId: QualifiedID? = intent.getStringExtra(EXTRA_RECEIVER_USER_ID)?.toQualifiedID(qualifiedIdMapper)
-            val sessionScope =
-                if (userId != null) {
-                    coreLogic.getSessionScope(userId)
-                } else {
-                    val currentSession = coreLogic.globalScope { session.currentSession() }
-                    if (currentSession is CurrentSessionResult.Success) {
-                        coreLogic.getSessionScope(currentSession.accountInfo.userId)
-                    } else {
-                        null
+            val userId: UserId? = intent.getStringExtra(EXTRA_RECEIVER_USER_ID)?.toQualifiedID(qualifiedIdMapper)
+                ?: coreLogic.globalScope {
+                    when (val currentSession = session.currentSession()) {
+                        is CurrentSessionResult.Success -> currentSession.accountInfo.userId
+                        else -> null
                     }
                 }
-
-            sessionScope?.let {
-                it.calls.rejectCall(qualifiedIdMapper.fromStringToQualifiedID(conversationId))
+            userId?.let {
+                coreLogic.getSessionScope(userId).calls.rejectCall(conversationIdString.toQualifiedID(qualifiedIdMapper))
+                callNotificationManager.hideIncomingCallNotification(userId.toString(), conversationIdString)
             }
-            CallNotificationManager.hideIncomingCallNotification(context)
         }
     }
 
     companion object {
```

Comment:
Just general thought. I like to build classes around intents to make them more structural and typed. It would be nice to have class representing different broadcasts intents. Something like
```kotlin
    class DeclineIncomingCallReciverIntent {

        val conversationId: String
        val receiverUserId: String

        constructor(conversationId: String, receiverUserId: String) {
            this.conversationId = conversationId
            this.receiverUserId = receiverUserId
        }

        companion object {
            private const val EXTRA_CONVERSATION_ID = "conversation_id_extra"
            private const val EXTRA_RECEIVER_USER_ID = "user_id_extra"

            fun newIntent(context: Context, conversationId: String, userId: String): Intent =
                Intent(context, DeclineIncomingCallReceiver::class.java).apply {
                    putExtra(EXTRA_CONVERSATION_ID, conversationId)
                    putExtra(EXTRA_RECEIVER_USER_ID, userId)
                }
            
            fun factory(intent: Intent): DeclineIncomingCallReciverIntent? {
                val conversationId = intent.getStringExtra(EXTRA_CONVERSATION_ID)
                val receiverUserId = intent.getStringExtra(EXTRA_RECEIVER_USER_ID)
                if (conversationId == null || receiverUserId == null) {
                  return null
                }
                return DeclineIncomingCallReciverIntent(conversationId, receiverUserId)
            }
        }
    }
```

Reply:
Collective is the best place to present ideas like this, it's definitely worth discussing